homeassistant/components/wilight/cover.py

```diff
@@ -0,0 +1,113 @@
+"""Support for WiLight Cover."""
+
+from pywilight.const import (
+    COVER_V1,
+    DOMAIN,
+    ITEM_COVER,
+    WL_CLOSE,
+    WL_CLOSING,
+    WL_OPEN,
+    WL_OPENING,
+    WL_STOP,
+    WL_STOPPED,
+)
+
+from homeassistant.components.cover import ATTR_POSITION, CoverEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+
+from . import WiLightDevice
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities
+):
+    """Set up WiLight covers from a config entry."""
+    parent = hass.data[DOMAIN][entry.entry_id]
+
+    # Handle a discovered WiLight device.
+    entities = []
+    for item in parent.api.items:
+        if item["type"] != ITEM_COVER:
+            continue
+        index = item["index"]
+        item_name = item["name"]
+        if item["sub_type"] != COVER_V1:
+            continue
+        entity = WiLightCover(parent.api, index, item_name)
+        entities.append(entity)
+
+    async_add_entities(entities)
+
+
+def wilight_to_hass_position(value):
+    """Convert wilight position 1..255 to hass format 0..100."""
+    return min(100, round((value * 100) / 255))
+
+
+def hass_to_wilight_position(value):
+    """Convert hass position 0..100 to wilight 1..255 scale."""
+    return min(255, round((value * 255) / 100))
+
+
+class WiLightCover(WiLightDevice, CoverEntity):
+    """Representation of a WiLights cover."""
+
+    def __init__(self, api_device, index, item_name):
+        """Initialize the device."""
+        super().__init__(api_device, index, item_name)
+        # Initialize the WiLights cover.
+        self._motor_state = WL_STOPPED
+        self._position = 100
+        self._target = 100
+
+    @property
+    def current_cover_position(self):
+        """Return current position of cover.
+
+        None is unknown, 0 is closed, 100 is fully open.
+        """
+        if "position_current" in self._status:
+            self._position = wilight_to_hass_position(self._status["position_current"])
+        return self._position
```

Comment:
```suggestion
        if "position_current" in self._status:
            return wilight_to_hass_position(self._status["position_current"])
        return None
```

If we do not know the position we should return `None`